### DISCUSSION TITLE
IdentityServer4.Endpoints.TokenEndpoint: Warning: Invalid HTTP request for token endpoint

### DISCUSSION THREAD

**Discussion by vikramvee** [kgrzybek/modular-monolith-with-ddd#306] [Q&A]
While sending the request to get the token, to the `http://localhost:5000/connect/token` api I get the error.
`IdentityServer4.Endpoints.TokenEndpoint: Warning: Invalid HTTP request for token endpoint`

I did a POST request with the below body

```
grant_type=password
&username=testMember@mail.com
&password=testMemberPass
&client_id=ro.client
&client_secret=secret
&scope=myMeetingsAPI
```

**Comment by vonvenera** [comment]
Are you using Frontend project to log in (https://github.com/kgrzybek/modular-monolith-with-ddd-fe-react)? 
Or are you running .http files included in /Api/RequestExamples?

When I try logging in with frontend app I get same error, but http files work ok.
The problem with frontend app is that it is trying to POST with **multipart/form-data** content type, while IdnentityServer4 TokenEndpoint expects **application/x-www-form-urlencoded** (see https://github.com/IdentityServer/IdentityServer4/blob/4dc10e665f5ede63274427036c25cdc216130eb9/src/IdentityServer4/src/Extensions/HttpRequestExtensions.cs#L39)

Not sure how to force frontend app to submit correct content type.
